lib/logstash/filters/aggregate.rb

```diff
@@ -371,23 +481,26 @@ def flush(options = {})
     if (@@eviction_instance == self && (@@last_eviction_timestamp.nil? || Time.now > @@last_eviction_timestamp + @timeout / 2))
       events_to_flush = remove_expired_maps()
       @@last_eviction_timestamp = Time.now
+      return events_to_flush
     end
-    
-    return events_to_flush
+
   end
 
   
   # Remove the expired Aggregate maps from @@aggregate_maps if they are older than timeout.
-  # If @push_previous_map_as_event option is set, expired maps are returned as new events to be flushed to Logstash pipeline.
+  # If @push_previous_map_as_event option is set, or @push_map_as_event_on_timeout is set, expired maps are returned as new events to be flushed to Logstash pipeline.
   def remove_expired_maps()
     events_to_flush = []
     min_timestamp = Time.now - @timeout
     
     @@mutex.synchronize do
+
       @@aggregate_maps.delete_if do |key, element| 
         if (element.creation_timestamp < min_timestamp)
-          if (@push_previous_map_as_event)
-            events_to_flush << LogStash::Event.new(element.map)
+          if (@push_previous_map_as_event) || (@push_map_as_event_on_timeout)
+
+            events_to_flush << create_timeout_event(element.map, key)
+
           end
```

Comment:
OK, this is clearly not important, but can you remove blank lines 501 and 503 ?
You have right to say I am painful :)